src/edu/washington/escience/myria/operator/CSVFileScanFragment.java

```diff
@@ -138,31 +143,81 @@ protected TupleBatch fetchNextReady() throws IOException, DbException {
         throw new DbException("Error parsing row " + lineNumber, e);
       }
       CSVRecord record = iterator.next();
+      // If the number of columns matches the schema for the first line, mark as discarded
+      if (record.size() == schema.numColumns() && lineNumber - 1 == 0 && partitionStartByteRange != 0) {
+        // Check if the previous 2 bytes contains a new line -- this should be a rare case
+        InputStreamReader startStreamReader =
+            new InputStreamReader(source.getInputStream(partitionStartByteRange - 1, partitionStartByteRange));
```

Comment:
Can we just initialize the `InputStream` to `(partitionStartByteRange - 1, partitionEndByteRange)` and avoid an extra round trip to S3? Then after reading the first byte (to check for a newline character), the reader will be initialized to begin at the second character, i.e. the beginning of the range. (Note that we are assuming an encoding here, i.e. that a newline character always maps to a single byte, but I think it's safe to assume that UTF-8 is our only supported encoding.)